server/rollback_runner.go

```diff
@@ -0,0 +1,161 @@
+package server
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sync"
+
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+
+	"github.com/bytebase/bytebase/api"
+	"github.com/bytebase/bytebase/common/log"
+	"github.com/bytebase/bytebase/plugin/db"
+	"github.com/bytebase/bytebase/plugin/db/mysql"
+	"github.com/bytebase/bytebase/store"
+)
+
+// NewRollbackRunner creates a new rollback runner.
+func NewRollbackRunner(store *store.Store, pgInstanceDir, dataDir string) *RollbackRunner {
+	return &RollbackRunner{
+		store:          store,
+		pgInstanceDir:  pgInstanceDir,
+		dataDir:        dataDir,
+		generateSignal: make(chan struct{}, 1),
+	}
+}
+
+// RollbackRunner is the rollback runner generating rollback SQL statements.
+type RollbackRunner struct {
+	store          *store.Store
+	pgInstanceDir  string
+	dataDir        string
+	generateSignal chan struct{}
+	generateMap    sync.Map
+}
+
+// Run starts the rollback runner.
+func (r *RollbackRunner) Run(ctx context.Context, wg *sync.WaitGroup) {
+	defer wg.Done()
+	r.retryGenerateRollbackSQL(ctx)
+	for {
+		select {
+		case <-r.generateSignal:
+			log.Debug("Received signal for generating rollback SQL.")
+			r.generateMap.Range(func(key, value any) bool {
```

Comment:
When is the task evicted from the generateMap?

Reply:
Oops...